code/modules/assembly/infrared.dm

```diff
@@ -127,47 +131,36 @@
 		visible = !(visible)
 		if(first)
 			first.vis_spread(visible)
+	if(href_list["turn"])
+		var/setdir = input("Select the infrared laser direction:") as null|anything in list("north","east","south","west")
+		switch(setdir)
+			if("north")
```

Comment:
i swear i saw a proc that did this switch thing itself..but i don't remember the name..crap